internal/encryption/encryptor.go

```diff
@@ -42,21 +45,23 @@ func generateEncryptionKey(_, _ string) ([]byte, error) {
 // generateTestEncryptionKey generates a deterministic encryption key for testing.
 // While testing, we also want to make sure different keys are generated for different docs and fields
 // and that's why we use the docID and fieldName to generate the key.
-func generateTestEncryptionKey(docID, fieldName string) ([]byte, error) {
-	return []byte(fieldName + docID + testEncryptionKey)[0:keyLength], nil
+func generateTestEncryptionKey(docID string, fieldName immutable.Option[string]) ([]byte, error) {
+	return []byte(fieldName.Value() + docID + testEncryptionKey)[0:keyLength], nil
 }
 
 // DocEncryptor is a document encryptor that encrypts and decrypts individual document fields.
 // It acts based on the configuration [DocEncConfig] provided and data stored in the provided store.
 // It uses [core.EncStoreDocKey] to store and retrieve encryption keys.
 type DocEncryptor struct {
-	conf  immutable.Option[DocEncConfig]
-	ctx   context.Context
-	store datastore.DSReaderWriter
+	conf          immutable.Option[DocEncConfig]
+	ctx           context.Context
+	store         datastore.DSReaderWriter
+	cache         map[core.EncStoreDocKey][]byte
```

Comment:
todo: This is an infinite cache?

Reply:
Update: Should still just delete the cache though ;)

Reply:
deleted cache